### PR TITLE
[orange-reputation-based-voting] add orange-reputation-based-voting

### PR DESCRIPTION
**Pull request by joeuan**
add orange-reputation-based-voting for orange-protocol

### CODE REVIEW

File: src/strategies/orange-reputation-based-voting/index.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export const author = 'orange-protocol';
+export const version = '0.1.0';
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
```
Comment (ChaituVR) [inline src/strategies/orange-reputation-based-voting/index.ts:11]: 'options' is defined but never used 🤔 but I see address and decimals being passed from examples.json?

File: src/strategies/orange-reputation-based-voting/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  snapshot
```
Comment (ChaituVR) [inline src/strategies/orange-reputation-based-voting/index.ts:12]: We need to pass snapshot to API and API should make sure it returns data based on the blocknumber else the results would be changing in future

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/orange-reputation-based-voting/index.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+): Promise<Record<string, number>> {
+  const query = `{
+    getBasedVotingStrategy(
+      addrs: ${JSON.stringify(addresses)},
+      space: "${space}",
+      snapshot: "${snapshot}",
+      network: "${network}",
+      provider: "${provider}",
```
Comment (bonustrack) [inline src/strategies/orange-reputation-based-voting/index.ts:20]: Provider is not a string or number, I assume it wont work well here, you'd rather not having it at all

Comment (ChaituVR) [reply src/strategies/orange-reputation-based-voting/index.ts:20]: @bonustrack This is input for the oraganprotocol API https://api.orangeprotocol.io/orange2c/query
I can't find their documentation or schema 🤔

Comment (bonustrack) [reply src/strategies/orange-reputation-based-voting/index.ts:20]: But this return the variable from the strategy with Ethers.js provider https://github.com/snapshot-labs/snapshot-strategies/pull/254/files#diff-b60e2ce70275e1efd45ba615c092306ebd2de4585a540f539a16f1ba91ed4ed0R9

**Review by bonustrack — APPROVED** [review]
Looks good thanks!

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**